src/com/hms/flexyosisoftconnector/OSIsoftServer.java

```diff
@@ -336,39 +342,38 @@ public int initTags() {
   private void initOMF() {
     // setup type
     String messageTypeHeader = "&messagetype=type";
-    final String responseFilename = "/usr/response.json";
 
     RequestHTTPS(
         OSIsoftConfig.getOmfUrl(),
         "POST",
         OSIsoftConfig.getOmfPostHeaders() + messageTypeHeader,
         PayloadBuilder.getStringTypeBody(),
         "",
-        responseFilename);
+        responseDirectory + OsisoftJsonPayload.STRING_TAG_TYPE + responseFileName);

```

Comment:
Is there any reason to build theses paths besides the 
```
responseDirectory + "containers" + currentTagIndex + responseFileName
```
one at runtime, you should just create static finals for these so there is no chance you are doing extra string building during runtime.

Reply:
example:

```
static final String stringResponseFilePath = responseDirectory + OsisoftJsonPayload.STRING_TAG_TYPE + responseFileName;